chalicelib/decorators.py

```diff
@@ -7,4 +25,52 @@ def wrapper(self, table_name: str, *args, **kwargs):
 
         return func(self, table_name, *args, **kwargs)
 
-    return wrapper
\ No newline at end of file
+    return wrapper
+
+
+def auth(blueprint, roles):
+    """
+    Decorator for authenticating and authorizing access to API routes.
+
+    Args:
+        blueprint (object): The Chalice Blueprint object, providing access to the current request.
+        role (list[str]): The required role for authorization.
+
+    Returns:
+        function: A decorator function that authenticates and authorizes access based on the provided role.
+
+    Raises:
+        401 Unauthorized: If the Authorization header is missing, the token is invalid, or the token has expired.
+        403 Forbidden: If the decoded role is not part of the given role.
+    """
+
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            api_request = blueprint.current_request
+            auth_header = api_request.headers.get("Authorization", None)
+            if not auth_header:
+                raise UnauthorizedError("Authorization header is missing.")
+
+            _, token = auth_header.split(" ", 1) if " " in auth_header else (None, None)
+            if not token:
+                raise UnauthorizedError("Token is missing.")
+
+            try:
+                ssm_client = boto3.client("ssm")
+                auth_secret = ssm_client.get_parameter(
+                    Name="/Zap/AUTH_SECRET", WithDecryption=True
+                )["Parameter"]["Value"]
+                decoded = jwt.decode(token, auth_secret, algorithms=["HS256"])
+                print(roles)
+                # TODO: if decoded role is not part of given, reject auth
```

Comment:
omg i thought i was being silly bc i couldn't figure out where u were actually doing something with the roles but looks like it's a to-do